File: extensions/nexus-forum/src/Model/DeviceSignal.php
<?php

namespace Nexus\Forum\Model;

use Flarum\Database\AbstractModel;
use Flarum\User\User;

class DeviceSignal extends AbstractModel
{
    protected $table = 'nexus_device_signals';

    public $timestamps = false;

    protected $dates = [
        'created_at',
        'expires_at',
    ];

    protected $casts = [
        'bluetooth_seen' => 'bool',
        'shake_detected' => 'bool',
        'gyro_available' => 'bool',
        'accuracy_m' => 'int',
    ];

    public function user()
    {
        return $this->belongsTo(User::class, 'user_id');
    }
}

File: server/src/main/java/nexus/campus/device/entity/DeviceSignal.java
package nexus.campus.device.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import nexus.campus.common.entity.User;

import java.time.LocalDateTime;

@Entity
@Table(name = "nexus_device_signals")
@Getter @Setter @NoArgsConstructor
public class DeviceSignal {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false, length = 80)
    private String purpose = "linkgo";

    @Column(name = "coarse_geohash", length = 32)
    private String coarseGeohash;

    @Column(name = "accuracy_m")
    private Integer accuracyM;

    @Column(name = "bluetooth_seen")
    private boolean bluetoothSeen;

    @Column(name = "shake_detected")
    private boolean shakeDetected;

    @Column(name = "gyro_available")
    private boolean gyroAvailable;

    @Column(columnDefinition = "text")
    private String payload;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "expires_at")
    private LocalDateTime expiresAt;

    @PrePersist
    protected void onCreate() { if (createdAt == null) createdAt = LocalDateTime.now(); }
}
